althea_kernel_interface/src/link_local_tools.rs

```diff
@@ -43,6 +43,23 @@ impl KernelInterface {
         }
     }
 
+    pub fn get_global_device_ip_v4(&self, dev: &str) -> Result<Ipv4Addr, Error> {
+        let output = self.run_command("ip", &["addr", "show", "dev", dev, "scope", "global"])?;
+        trace!("Got {:?} from `ip addr`", output);
+
+        let re = Regex::new(r"inet (\S*?)(/[0-9]+)? scope global").unwrap();
+        let str = String::from_utf8(output.stdout)?;
```

Comment:
Rust is clever enough to understand this, but using `str` is pretty risky. Even the syntax highlighting assumes it to be the keyword for the type and not a local var name you can use.

Reply:
yeha changed that.